tests/network/bindingplugin_passt.go

```diff
@@ -133,61 +172,106 @@ var _ = Describe(SIG(" VirtualMachineInstance with passt network binding plugin"
+					Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed())
+
+					By("Connecting from the client VM")
+					Expect(console.RunCommand(clientVMI, connectToServerCmd(serverIP, lowTCPPort), 30*time.Second)).To(Succeed())
+
+					By("Connecting from the client VM to a port not specified on the VM spec")
+					Expect(console.RunCommand(clientVMI, connectToServerCmd(serverIP, lowTCPPort+1), 30)).NotTo(Succeed())
 				},
-					Entry("with a specific port number [IPv4]", []v1.Port{{Name: "http", Port: 8080, Protocol: "TCP"}}, 8080, k8sv1.IPv4Protocol),
-					Entry("with a specific lower port number [IPv4]", []v1.Port{{Name: "http", Port: 80, Protocol: "TCP"}}, 80, k8sv1.IPv4Protocol),
-					Entry("without a specific port number [IPv4]", []v1.Port{}, 8080, k8sv1.IPv4Protocol),
-					Entry("with a specific port number [IPv6]", []v1.Port{{Name: "http", Port: 8080, Protocol: "TCP"}}, 8080, k8sv1.IPv6Protocol),
-					Entry("without a specific port number [IPv6]", []v1.Port{}, 8080, k8sv1.IPv6Protocol),
+					Entry("[IPv4]", k8sv1.IPv4Protocol),
+					Entry("[IPv6]", k8sv1.IPv6Protocol),
 				)
 			})
 
 			Context("UDP", func() {
-				DescribeTable("Client server connectivity", func(ipFamily k8sv1.IPFamily) {
-					libnet.SkipWhenClusterNotSupportIPFamily(ipFamily)
+				var clientVMI *v1.VirtualMachineInstance
+				var serverVMI *v1.VirtualMachineInstance
 
-					const SERVER_PORT = 1700
-					var ports = []v1.Port{{Port: SERVER_PORT, Protocol: "UDP"}}
+				const udpPort = 1700
+
+				BeforeEach(func() {
+					var ports = []v1.Port{{Port: udpPort, Protocol: "UDP"}}
 
 					By("Starting server VMI")
-					serverVMI := libvmifact.NewAlpineWithTestTooling(
+					serverVMI = libvmifact.NewAlpineWithTestTooling(
 						libvmi.WithInterface(libvmi.InterfaceWithPasstBindingPlugin(ports...)),
 						libvmi.WithNetwork(v1.DefaultPodNetwork()),
 					)
 					serverVMI, err = kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), serverVMI, metav1.CreateOptions{})
 					Expect(err).ToNot(HaveOccurred())
 
 					By("Starting client VMI")
-					clientVMI := libvmifact.NewAlpineWithTestTooling(
+					clientVMI = libvmifact.NewAlpineWithTestTooling(
 						libvmi.WithInterface(libvmi.InterfaceWithPasstBindingPlugin()),
 						libvmi.WithNetwork(v1.DefaultPodNetwork()),
 					)
 					clientVMI, err = kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), clientVMI, metav1.CreateOptions{})
 					Expect(err).ToNot(HaveOccurred())
 
 					waitUntilVMIsReady(console.LoginToAlpine, serverVMI, clientVMI)
+				})
+				DescribeTable("Client server connectivity", func(ipFamily k8sv1.IPFamily) {
+					libnet.SkipWhenClusterNotSupportIPFamily(ipFamily)
 
 					By("Starting a UDP server")
-					vmnetserver.StartPythonUDPServer(serverVMI, SERVER_PORT, ipFamily)
+					vmnetserver.StartPythonUDPServer(serverVMI, udpPort, ipFamily)
```

Comment:
In the TCP scenario, we start the server on the `BeforeEach` stanza, is there a reason to not do it here as well?

Reply:
The TCP case does not have the `ipFamily` as an input.
It relates to how the UDP server is created. I have not found a nice way to remove the dependency on the family and did not invest in it further.